tutorials/03_physics_plugins.md

```diff
@@ -0,0 +1,163 @@
+| LinkFeatures  | applies external force and torque to link  |
+| SDFFeatures  | constructs entities from SDF file  |
+| ShapeFeatures  | retrieves `Shape` related properties like `BoundingBox`, `ShapeSize` etc. |
+| SimulationFeatures  | updates `World` and everything within by defined stepsize |
+
+### Dart vs. TPE vs. Bullet
+
+<!-- TODO: add Bullet once it's supported -->
+<!-- ### Bullet -->
+
+Dart ([Dynamic Animation and Robotics Toolkit](https://dartsim.github.io/)) is an open source library that provides data structures and algorithms for kinematic and dynamic applications in robotics and computer animation.
+It is the default physics engine used in Ignition Simulation.
+The source code for Dartsim plugin can be found in [Ignition Physics repository](https://github.com/ignitionrobotics/ign-physics/tree/master) under `dartsim` directory.
+
+TPE ([Trivial Physics Engine](<!-- add repo link after merged-->)) is an open source library created by Open Robotics that enables fast, inexpensive kinematics simulation for entities at large scale.
+It supports higher-order fleet dynamics without real physics (eg. gravity, force, constraint etc.) and multi-machine synchronization.
+Ignition support for TPE targets [Citadel](https://ignitionrobotics.org/docs/citadel) and onward releases.
+The source code for TPE plugin can be found in [Ignition Physics repository](https://github.com/ignitionrobotics/ign-physics/tree/ign-physics2) under `tpe/plugin` directory
+
+Bullet ([Bullet Real-Time Physics Simulation](https://github.com/bulletphysics/bullet3)) is a physics engine that simulates collision detection, soft and rigid body dynamics.
+Ignition support for Bullet targets [Blueprint](https://ignitionrobotics.org/docs/blueprint) and onward releases.
+The source code for bullet plugin can be found in [Ignition Physics repository](https://github.com/ignitionrobotics/ign-physics/tree/ign-physics1) under `bullet` directory.
+
+The following is a list of features supported by each physics engine to help users select one that fits their needs.
+
+#### Entity Comparison
+
+The following is a table of `Entity` names used in Ignition Physics plugin interface, Dart and TPE.
+Entities are arranged in top-down hierarchical order.
+
+| Physics Plugin | Dart  | TPE  | Bullet |
+|:-:|:-:|:-:|:-:|
+| Engine  | Engine  | Engine  | Engine |
+| World  | World  | World  | MultiBodyDynamicsWorld |
+| Frame  | Frame  | N/A  | N/A |
+| Model  | Skeleton  | Model | MultiBody |
+| Joint  | Joint  | N/A | MultiBodyJoint |
+| Link  | BodyNode  | Link | Scalar |
+| Shape  | Shape  | Collision | CollisionShape |
+| Box/Sphere/Cylinder etc. | Box/Sphere/Cylinder etc. | Box/Sphere/Cylinder/Mesh etc. | Box/TriangleMesh/Convex etc. |
+
+#### Feature Comparison
+
+The following is a table of implemented `Features` of Dartsim and TPE-Plugin.
+
+| Features | Dartsim | TPE-Plugin | Bullet |
+|:-:|:-:|:-:|:-:|
+| Base | ✓ | ✓ | ✓ |
+| RetrieveWorld | ✓  | ✓ | ✕ |
```

Comment:
`RetrieveWorld` is a DART specific (maybe also TPE specific) feature used for testing internally. I'm not sure if it should be mentioned here.